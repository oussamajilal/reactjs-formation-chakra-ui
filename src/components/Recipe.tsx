import { Box, Button } from "@chakra-ui/react";
import * as React from "react"

interface RecipeProps {
    id: number;
}

const Recipe = (props: RecipeProps) => {
    const [inc, setInc] = React.useState(0);
    const handleIncClick = () => {
        setInc(inc + 1);
    };

    return <Box>
        <span>
            My Recipe NÂ° {props.id} incremented
            <span data-testid="recipe-inc">{inc}</span> times
        </span>
        <Button data-testid="button-inc" onClick={handleIncClick}>+</Button>
    </Box>
}

export default Recipe;