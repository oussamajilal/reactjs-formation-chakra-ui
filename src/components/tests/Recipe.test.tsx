import React from "react"
import { cleanup } from '@testing-library/react'
import { render } from "../../test-utils"
import Recipe from "../Recipe"

afterEach(cleanup)

describe('Recipe', () => {
    it('should render correctly', () => {
        const { asFragment } = render(<Recipe id={1} />)
        expect(asFragment()).toMatchSnapshot()
    })

    it('should increment the counter when clicking the button', () => {
        const { getByTestId } = render(<Recipe id={1} />)
        const recipeInc = getByTestId('recipe-inc')
        const buttonInc = getByTestId('button-inc')
        expect(recipeInc).toHaveTextContent('0')
        buttonInc.click()
        expect(recipeInc).toHaveTextContent('1')
        buttonInc.click()
        expect(recipeInc).toHaveTextContent('2')
        buttonInc.click()
        expect(recipeInc).toHaveTextContent('3')
    })
})