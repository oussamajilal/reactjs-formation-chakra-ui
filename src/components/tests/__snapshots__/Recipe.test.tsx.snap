// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-0"
  >
    <span>
      My Recipe NÂ° 1 incremented
      <span
        data-testid="recipe-inc"
      >
        0
      </span>
       times
    </span>
    <button
      class="chakra-button css-taj3dd"
      data-testid="button-inc"
      type="button"
    >
      +
    </button>
  </div>
</DocumentFragment>
`;
